wasm/wat/token.go

```diff
@@ -0,0 +1,217 @@
+	//
+	// For example, in the below, 'local.get' 'i32.const' and 'i32.lt_s' are keywords:
+	//		local.get $y
+	//		i32.const 6
+	//		i32.lt_s
+	//
+	// See https://www.w3.org/TR/wasm-core-1/#text-keyword
+	tokenKeyword
+
+	// tokenUN is an unsigned integer in decimal or hexadecimal notation, optionally separated by underscores.
+	//
+	// For example, the following tokens represent the same number: 10
+	//		(i32.const 10)
+	//		(i32.const 1_0)
+	//		(i32.const 0x0a)
+	//		(i32.const 0x0_A)
+	//
+	// See https://www.w3.org/TR/wasm-core-1/#text-int
+	tokenUN
+
+	// tokenSN is a signed integer in decimal or hexadecimal notation, optionally separated by underscores.
+	//
+	// For example, the following tokens represent the same number: 10
+	//		(i32.const +10)
+	//		(i32.const +1_0)
+	//		(i32.const +0x0a)
+	//		(i32.const +0x0_A)
+	//
+	// See https://www.w3.org/TR/wasm-core-1/#text-int
+	tokenSN
+
+	// tokenFN represents an IEEE-754 floating point number in decimal or hexadecimal notation, optionally separated by
+	// underscores. This also includes special constants for infinity ('inf') and NaN ('nan').
+	//
+	// For example, the right-hand side of the following S-expressions are all valid floating point tokens:
+	//		(f32.const +nan)
+	//		(f64.const -nan:0xfffffffffffff)
+	//		(f64.const -inf)
+	//      (f64.const +0x0.0p0)
+	//      (f32.const 0.0e0)
+	//		(i32.const +0x0_A)
+	//		(f32.const 1.e10)
+	//      (f64.const 0x1.fff_fff_fff_fff_fp+1_023)
+	//		(f64.const 1.7976931348623157e+308)
+	tokenFN
+
+	// tokenString is a UTF-8 sequence enclosed by quotation marks, representing an encoded byte string. A tokenString
+	// can contain any character except ASCII control characters, quotation marks ('"') and backslash ('\'): these must
+	// be escaped. tokenString characters correspond to UTF-8 encoding except the special case of '\hh', which allows
+	// raw bytes expressed as hexadecimal.
+	//
+	// For example, the following tokens represent the same raw bytes: 0xe298ba0a
+	//		(data (i32.const 0) "☺\n")
+	//		(data (i32.const 0) "\u{263a}\u{0a}")
+	//		(data (i32.const 0) "\e2\98\ba\0a")
+	//
+	// See https://www.w3.org/TR/wasm-core-1/#strings%E2%91%A0
+	tokenString
+
+	// tokenId is a sequence of idChar characters prefixed by a '$':
+	//
+	// For example, in the below, '$y' is an id:
+	//		local.get $y
+	//		i32.const 6
+	//		i32.lt_s
+	//
+	// See https://www.w3.org/TR/wasm-core-1/#text-id
+	tokenId
+
+	// tokenLParen is a left paren: '('
+	tokenLParen
+
+	// tokenRParen is a right paren: ')'
+	tokenRParen
+
+	// tokenReserved is a sequence of idChar characters which are neither a tokenId nor a tokenString.
+	//
+	// For example, '0$y' is a tokenReserved, because it doesn't start with a letter or '$'.
+	//
+	// See https://www.w3.org/TR/wasm-core-1/#text-reserved
+	tokenReserved
+)
+
+// tokenNames is index-coordinated with tokenType
+var tokenNames = []string{
```

Comment:
nit 
```suggestion
var tokenNames = [...]string{
```

Reply:
oh cool. my go is better when you teach me things like this!